src/components/Sidebar/Sidebar.tsx

```diff
@@ -109,15 +173,23 @@ const Sidebar: FC<SidebarProps> = ({ headerElement }) => {
         <div className={styles.sidebar}>
           <SidebarHeader />
           <hr />
-          <DropDown
-            label={'Node Type'}
-            id={'Node Type'}
-            options={[
-              { label: 'Action', value: 'action' },
-              { label: 'Branch', value: 'branch' }
-            ]}
-          />
-          <AddNodes />
+          <FormControl variant="outlined" className={classes.formControl} fullWidth>
+            <InputLabel>Node Type</InputLabel>
+            <Select
+              value={currentNode.nodeType}
+              onChange={handleTypeChange}
+              label="Node Type"
+              error={currentNode.nodeType === null}
+            >
+              <MenuItem value={'action'}>Action</MenuItem>
+              <MenuItem value={'branch'}>Branch</MenuItem>
+            </Select>
+          </FormControl>
+          {currentNode.nodeType === 'branch' ? (
+            <AddBranchNode currentNode={currentNode} addChoiceNode={addChoiceNoide} />
+          ) : (
+            <AddNodes addBranchNode={addBranchNode} />
+          )}
```

Comment:
I think the naming is a little confusing here. If the current node is branch then we can add choice (transition nodes) but otherwise we show a button to add a new branch node right? But the component name in the first case is `AddBranchNode`

Reply:
If we renamed the component to `BranchNode` would that make it more clear?

Reply:
Renamed to `BranchNode`.